neuralprophet/TorchProphet.py

```diff
@@ -0,0 +1,553 @@
+            log.info(
+                "Using `_prior_scale` is unsupported for regularisation in NeuralProphet, please use the `_reg` args instead."
+            )
+        if mcmc_samples or uncertainty_samples:
+            log.info(
+                "Providing the number of samples for Bayesian inference or Uncertainty estimation is not required in NeuralProphet."
+            )
+        if stan_backend:
+            log.info("A stan_backend is not used in NeuralProphet. Please remove the parameter")
+        # Run the NeuralProphet function
+        super(TorchProphet, self).__init__(
+            growth=growth,
+            changepoints=changepoints,
+            n_changepoints=n_changepoints,
+            changepoints_range=changepoint_range,
+            yearly_seasonality=yearly_seasonality,
+            weekly_seasonality=weekly_seasonality,
+            daily_seasonality=daily_seasonality,
+            seasonality_mode=seasonality_mode,
+            # TODO: activate once PR merged prediction_interval=interval_width,
+            **kwargs,
+        )
+        # Handle holidays as events
+        if holidays is not None:
+            self.add_events(
+                events=list(holidays["holiday"].unique()),
+                lower_window=holidays["lower_window"].max(),
+                upper_window=holidays["upper_window"].max(),
+            )
+            self.events_df = holidays.copy()
+            self.events_df.rename(columns={"holiday": "event"}, inplace=True)
+            self.events_df.drop(["lower_window", "upper_window"], axis=1, errors="ignore", inplace=True)
+
+        # Overwrite NeuralProphet properties
+        self.name = "Prophet"
+        self.history = None
+
+        # Unused properties
+        self.train_holiday_names = None
+
+    def validate_inputs(self):
+        """
+        Validates the inputs to Prophet.
+        """
+        raise NotImplementedError("Not required in NeuralProphet as all inputs are automatically checked.")
+
+    def validate_column_name(self, name, check_holidays=True, check_seasonalities=True, check_regressors=True):
+        """Validates the name of a seasonality, holiday, or regressor.
+
+        Parameters
+        ----------
+        name: string
+        check_holidays: bool check if name already used for holiday
+        check_seasonalities: bool check if name already used for seasonality
+        check_regressors: bool check if name already used for regressor
+        """
+        super(TorchProphet, self)._validate_column_name(
+            name=name,
+            events=check_holidays,
+            seasons=check_seasonalities,
+            regressors=check_regressors,
+            covariates=check_regressors,
+        )
+
+    def setup_dataframe(self, df, initialize_scales=False):
+        """
+        Dummy function that raises an error.
+
+        This function is not supported in NeuralProphet.
+        """
+        raise NotImplementedError(
```

Comment:
instead of an error, we could just log an error and move on, as this should not cause any issues?